azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -1043,6 +1057,17 @@ func expandKubernetesClusterAddonProfiles(d *schema.ResourceData) map[string]*co
 		}
 	}
 
+	azurePolicy := profile["azure_policy"].([]interface{})
+	if len(azurePolicy) > 0 {
+		value := azurePolicy[0].(map[string]interface{})
```

Comment:
`azurePolicy[0]` could be nil so we should do a nil check before casting it to `map[string]interface{}`

Reply:
sure - I kept it in line with the other policy addons, but I will add a nil check now.